netlogo-gui/src/main/ide/CodeCompletionPopup.scala

```diff
@@ -87,48 +91,66 @@ case class CodeCompletionPopup(compiler: CompilerServices) {
   def displayPopup(): Unit = {
     for{eA <- editorArea} {
       isPopupEnabled = true
-      val token = compiler.getTokenAtPosition(eA.getText(), eA.getCaretPosition)
-      val position = if (token != null) token.start else eA.getCaretPosition
+      val tokenOption = getTokenTillPosition(eA.getText(), eA.getCaretPosition)
+      var position = 0
+      tokenOption match {
+        case Some(token) => position = token.start
+        case None => position = eA.getCaretPosition
+      }
       window.setLocation(eA.getLocationOnScreen.x + eA.modelToView(position).x,
         eA.getLocationOnScreen.y + eA.modelToView(position).y + eA.getFont.getSize)
-      updatePopup()
+      updatePopup(None)
     }
   }
 
   /**
     * Updates the list of the suggestion box and makes it visible.
     */
-  def updatePopup(): Unit = {
-    println("I was called")
+  def updatePopup(docEventOption: Option[DocumentEvent]): Unit = {
     for{eA <- editorArea} {
-      if (isPopupEnabled) {
-        val token = compiler.getTokenAtPosition(eA.getText(), eA.getCaretPosition)
-        var list = Seq[String]()
-        // word only contains the current token till the cursor position
-        val word = if (token != null) token.text.substring(0, eA.getCaretPosition - token.start) else null
-        // popup not to be diplayed after user hits the enter key
-        if (word != null && !token.text.equals(lastSuggested)) {
-          list = autoSuggest.getSuggestions(word)
-          dlm.removeAllElements()
-          list.foreach(dlm.addElement(_))
-          if (!list.isEmpty) {
-            lastSuggested = ""
-            window.validate()
-            window.setVisible(true)
-            // Required to keep the caret in the editorArea visible
-            eA.getCaret.setVisible(true)
-            suggestionDisplaylist.setSelectedIndex(0)
-          } else {
-            isPopupEnabled = false
+      if (isPopupEnabled ) {
+        var position = eA.getCaretPosition
+        for{docEvent <- docEventOption} {
```

Comment:
Only lines 111-119 require the `docEventOption` parameter. Perhaps it would be better to extract a separate method. that calls into this one?